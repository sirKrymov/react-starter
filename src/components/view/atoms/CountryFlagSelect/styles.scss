@import '../../../../styles/mixins';

.country-flag-select {
  .flag-select {
    padding-bottom: 0;
    height: 36px;
    width: 40px;

    &__btn {
      padding: 0;
      height: 100%;
      width: 100%;

      &::after {
        display: none;
      }
    }

    &__option {
      margin: 0;
    }

    &__options {
      transform: translateX(-50%);
      margin: 0;
      width: 100%;
      left: 50%;
      background: transparent;
    }
  }
}
