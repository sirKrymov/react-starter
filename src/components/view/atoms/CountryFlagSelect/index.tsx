// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement } from 'react';

import cn from 'classnames';

import ReactFlagsSelect from 'react-flags-select';

import './styles.scss';

interface IProps {
  onSelect(country: string): void;
  defaultCountry?: string;
  customConfig?: Record<string, any>;
  countries?: string[];
  className?: string;
  id: string;
}

export function CountryFlagSelect({
  defaultCountry = 'GB',
  customConfig,
  className,
  countries = ['GB'],
  onSelect,
  id
}: IProps): ReactElement {
  const countryFlagSelectClass = cn(
    {
      'country-flag-select': true
    },
    className
  );

  return (
    <div className={countryFlagSelectClass} id={id}>
      <ReactFlagsSelect
        showSelectedLabel={false}
        showOptionLabel={false}
        defaultCountry={defaultCountry}
        countries={countries}
        onSelect={onSelect}
        {...customConfig}
      />
    </div>
  );
}
