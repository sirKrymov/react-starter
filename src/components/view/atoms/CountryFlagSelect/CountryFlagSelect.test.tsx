import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { CountryFlagSelect } from '../CountryFlagSelect';

describe('<CountryFlagSelect> atom component', () => {
  const handleSelect = jest.fn();

  let shallowComponent: ShallowWrapper;

  const createComponent = () => {
    shallowComponent = shallow(
      <CountryFlagSelect
        id="check-country-flag-select"
        onSelect={handleSelect}
      />
    );
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });
});
