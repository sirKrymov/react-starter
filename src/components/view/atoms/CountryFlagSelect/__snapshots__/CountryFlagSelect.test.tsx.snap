// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountryFlagSelect> atom component WHEN the component loads THEN it should match snapshot 1`] = `
<div
  className="country-flag-select"
  id="check-country-flag-select"
>
  <ReactFlagsSelect
    alignOptions="right"
    blackList={false}
    countries={
      Array [
        "GB",
      ]
    }
    customLabels={Object {}}
    defaultCountry="GB"
    disabled={false}
    onSelect={[MockFunction]}
    optionsSize={14}
    placeholder="Select a country"
    searchPlaceholder="Search"
    searchable={false}
    selectedSize={16}
    showOptionLabel={false}
    showSelectedLabel={false}
  />
</div>
`;
